### PR TITLE
feat: Update API version to support EntraID authentication to `avm/res/cache/redis`

### LINKED ISSUE #2190: [AVM Module Issue]: Support Entra ID authentication with Microsoft Cache for Redis
### Check for previous/existing GitHub issues

- [X] I have checked for previous/existing GitHub issues

### Issue Type?

Feature Request

### Module Name

avm/res/cache/redis

### (Optional) Module Version

0.3.2

### Description

Hey there! 👋 Long time user, first time caller.

Went to create a new Redis instance using the new [Entra ID authentication support](https://learn.microsoft.com/en-us/azure/azure-cache-for-redis/cache-whats-new#microsoft-entra-id), but the deployment ended up failing when providing `'aad-enabled': 'true'` in the `redisConfiguration` parameter. The error cites Microsoft Entra Authentication is supported only for requests with minimum API version 2023-05-01-preview, which makes sense as the current module is using 2022-06-01.

Now that Entra ID for cache authentication is out in GA, can this use case be supported? 🙏

### (Optional) Correlation Id

_No response_

### LINKED ISSUE #2204: [AVM Module Issue]: Update Redis version to 2024-03-01.
### Check for previous/existing GitHub issues

- [X] I have checked for previous/existing GitHub issues

### Issue Type?

Bug

### Module Name

avm/res/cache/redis

### (Optional) Module Version

0.3.2

### Description

When attempting to deploy a Redis cache using version 0.3.2 with the EntraID Auth enabled I get the following error:
```
Microsoft Entra Authentication is supported only for requests with minimum API version 2023-05-01-preview. Please update the API version for the request URL to 2023-05-01-preview or later versions and then retry the request.
```

To repo just set:
```json
redisConfiguration: {
"aad-enabled": true
}
```

The fix is to update the Redis resource to apiVersion  `2022-06-01`.

### (Optional) Correlation Id

_No response_

### PR DESCRIPTION
**Pull request by hundredacres**
## Description

Updating API version to support EntraID authentication

Fixes #2190
Fixes #2204 
Closes #2190 
Clsoes #2204 

## Pipeline Reference

| Pipeline |
| -------- |
|    [avm.res.cache.redis](https://github.com/hundredacres/bicep-registry-modules/actions/workflows/avm.res.cache.redis.yml)      |

## Type of Change

- [ ] Update to CI Environment or utlities (Non-module effecting changes)
- [X] Azure Verified Module updates:
  - [ ] Bugfix containing backwards compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
    - [X] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [ ] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [X] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [ ] Update to documentation

## Checklist

- [X] I'm sure there are no other open Pull Requests for the same update/change
- [X] I have run `Set-AVMModule` locally to generate the supporting module files.
- [X] My corresponding pipelines / checks run clean and green without any errors or warnings

### CODE REVIEW

File: avm/res/compute/virtual-machine/main.bicep
Comment (ChrisSidebotham) [inline avm/res/compute/virtual-machine/main.bicep:1]: Are these changes supposed to be in this PR? Can this be reverted?

Comment (hundredacres) [reply avm/res/compute/virtual-machine/main.bicep:1]: They are not meant to be in this PR. Will revert.

Comment (hundredacres) [reply avm/res/compute/virtual-machine/main.bicep:1]: They are not meant to be in this PR. Will revert.

Comment (hundredacres) [reply avm/res/compute/virtual-machine/main.bicep:1]: Removed.

**Review by eriqua — COMMENTED** [review]
Hey @hundredacres any reason for removing the Needs module owner label? Any member of the owner group can review and approve this PR, hence the label.

Cc @segraef, could you review once you get a moment? 🙂

**Comment by hundredacres** [comment]
No reason, just a mistake on my end.Sent from my iPhoneOn Jun 6, 2024, at 4:30 PM, Erika Gressi ***@***.***> wrote:﻿
@eriqua commented on this pull request.

Hey @hundredacres any reason for removing the Needs module owner label? Any member of the owner group can review and approve this PR, hence the label.
Cc @segraef, could you review once you get a moment? 🙂

**Comment by jrlsampension** [comment]
Any hope of this getting a final review so we can get the Entra support feature?

**Comment by hundredacres** [comment]
@segraef can you review this when you get a chance?

**Comment by hundredacres** [comment]
@Azure/bicep-admins Can I get an approval on this so we can merge? Been waiting for 3 weeks now.

**Review by AlexanderSehr — APPROVED** [review]

**Pull request merged**